c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -88,18 +88,15 @@ C2R MfxC2DecoderComponent::MaxPictureSizeSetter(bool mayBlock, C2P<C2StreamMaxPi
     // TODO: get max width/height from the size's field helpers vs. hardcoding
     me.set().width = c2_min(c2_max(me.v.width, size.v.width), uint32_t(WIDTH_8K));
     me.set().height = c2_min(c2_max(me.v.height, size.v.height), uint32_t(HEIGHT_8K));
-    ALOGD("zyc, max.w = %d, h = %d", me.set().width, me.set().height);
     return C2R::Ok();
 }
 
 C2R MfxC2DecoderComponent::MaxInputSizeSetter(bool mayBlock, C2P<C2StreamMaxBufferSizeInfo::input> &me,
                                 const C2P<C2StreamMaxPictureSizeTuning::output> &maxSize) {
     (void)mayBlock;
     // assume compression ratio of 2
-    me.set().value = c2_max((((maxSize.v.width + 15) / 16)
-            * ((maxSize.v.height + 15) / 16) * 192), kMinInputBufferSize);
-    ALOGD("zyc, input size = %d", me.set().value);
-    me.set().value = kMinInputBufferSize;
+    me.set().value = c2_max((((maxSize.v.width + 63) / 64) * ((maxSize.v.height + 63) / 64) * 3072),
```

Comment:
Can you please explain more for this change? Generally, the input buffer size is calculated with the assuming compression ration... no related to the buffer alignment.

Reply:
This code is copied from SW. In SW, HEVS is aligned with 64, AVC is aligned with 16. We need to support all formats. so we keep the size aligned with 64.